pkg/ccl/crosscluster/logical/dead_letter_queue_test.go

```diff
@@ -73,10 +78,85 @@ func TestLoggingDLQClient(t *testing.T) {
 
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
-			if tc.reason == nil {
-				tc.reason = errors.New("some error")
+			if tc.applyError == nil {
+				tc.applyError = errors.New("some error")
 			}
-			err := dlqClient.Log(ctx, tc.jobID, tc.kv, tc.cdcEventRow, tc.reason)
+			err := dlqClient.Log(ctx, tc.jobID, tc.kv, tc.cdcEventRow, tc.dlqReason)
+			if tc.expectedErrMsg == "" {
+				require.NoError(t, err)
+			} else {
+				require.ErrorContains(t, err, tc.expectedErrMsg)
+			}
+		})
+	}
+}
+
+func TestDLQClient(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+	ctx := context.Background()
+
+	srv, db, kvDB := serverutils.StartServer(t, base.TestServerArgs{})
+	defer srv.Stopper().Stop(context.Background())
+	s := srv.ApplicationLayer()
+
+	sd := sql.NewInternalSessionData(ctx, s.ClusterSettings(), "" /* opName */)
+	ie := s.InternalDB().(isql.DB).Executor(isql.WithSessionData(sd))
+
+	sqlDB := sqlutils.MakeSQLRunner(db)
+	sqlDB.Exec(t, `CREATE TABLE foo (a INT)`)
+
+	tableName := "foo"
+	tableDesc := desctestutils.TestingGetPublicTableDescriptor(kvDB, s.Codec(), "defaultdb", tableName)
+	familyDesc := &descpb.ColumnFamilyDescriptor{
+		ID:   descpb.FamilyID(1),
+		Name: "",
+	}
+
+	ed, err := cdcevent.NewEventDescriptor(tableDesc, familyDesc, false, false, hlc.Timestamp{})
+	require.NoError(t, err)
+
+	tableID := int32(tableDesc.GetID())
+	dlqClient := InitDeadLetterQueueClient(ie)
+	require.NoError(t, dlqClient.Create(ctx, []int32{tableID}))
+
+	var tableNameQueryResult string
+	sqlDB.QueryRow(t, `SELECT table_name FROM [SHOW TABLES FROM defaultdb]`).Scan(&tableNameQueryResult)
+	require.Equal(t, tableName, tableNameQueryResult)
+
+	type testCase struct {
+		name           string
+		expectedErrMsg string
+
+		jobID       int64
+		tableID     int32
+		tableName   string
+		kv          streampb.StreamEvent_KV
+		cdcEventRow cdcevent.Row
+		applyError  error
+		dlqReason   retryEligibility
+	}
+
+	testCases := []testCase{
+		{
+			name:        "insert row into dlq table",
+			cdcEventRow: cdcevent.Row{EventDescriptor: ed},
+			tableID:     tableID,
+			tableName:   tableName,
+			dlqReason:   noSpace,
+		},
+		{
+			name:           "expect error when given nil cdcEventRow",
+			expectedErrMsg: "cdc event row not initialized",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			if tc.applyError == nil {
+				tc.applyError = errors.New("some error")
+			}
+			err := dlqClient.Log(ctx, tc.jobID, tc.kv, tc.cdcEventRow, tc.dlqReason)
 			if tc.expectedErrMsg == "" {
 				require.NoError(t, err)
```

Comment:
either in this pr, or in the follow up, you should read the row you just inserted, to verify you inserted the data you expected to.

Reply:
Will include that in the folllowup pr! re: steven's comment above, will add more tests & refactoring in the followup pr as well.